### ISSUE TITLE
flexible array members support

### ISSUE THREAD

**Issue by yrashk** [Tsche/repr#24]
For this data structure:

```c++
struct some_person {
    std::string name;
    unsigned birth_year;
    int data[];
};
```

I am getting this error:

```
error: type 'some_person' decomposes into 3 elements, but only 2 names were provided
```

Normal array members work fine.

Do you think it would be possible to support these?

**Comment by yrashk** [comment]
Same applies to zero-sized arrays.

**Comment by yrashk** [comment]
As far as I can tell, it is failing because for zero-sized arrays the `requires` in the arity calculator is failing because of `initializer for aggregate with no elements requires explicit braces` and for flexible members array it fails because of `initialization of flexible array member is not allowed`

In both cases, a simple array initializer `{}` works.

**Comment by Tsche** [comment]
Good point, I'll try to find a workaround next time I'm messing with this library :)
